services/src/main/java/org/keycloak/keys/GeneratedRsaKeyProviderFactory.java

```diff
@@ -18,136 +18,69 @@
 package org.keycloak.keys;
 
 import org.jboss.logging.Logger;
-import org.keycloak.common.util.CertificateUtils;
-import org.keycloak.common.util.KeyUtils;
-import org.keycloak.common.util.MultivaluedHashMap;
-import org.keycloak.common.util.PemUtils;
 import org.keycloak.component.ComponentModel;
-import org.keycloak.component.ComponentValidationException;
 import org.keycloak.crypto.Algorithm;
 import org.keycloak.crypto.KeyUse;
 import org.keycloak.models.KeycloakSession;
-import org.keycloak.models.RealmModel;
-import org.keycloak.provider.ConfigurationValidationHelper;
 import org.keycloak.provider.ProviderConfigProperty;
 
-import java.security.KeyPair;
-import java.security.PrivateKey;
-import java.security.cert.Certificate;
-import java.security.interfaces.RSAPrivateKey;
 import java.util.List;
 
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
-public class GeneratedRsaKeyProviderFactory extends AbstractRsaKeyProviderFactory {
+public class GeneratedRsaKeyProviderFactory extends AbstractGeneratedRsaKeyProviderFactory {
 
     private static final Logger logger = Logger.getLogger(GeneratedRsaKeyProviderFactory.class);
 
     public static final String ID = "rsa-generated";
 
     private static final String HELP_TEXT = "Generates RSA keys and creates a self-signed certificate";
```

Comment:
Should the help text be updated to explicitly mention that it is used for "signatures" ? Ideally align it with the help text you used for new "enc" based provider

Reply:
@mposolda Yes, I will change the help text as follows.

`"Generates RSA signature keys and creates a self-signed certificate"`